homeassistant/components/harmony/select.py

```diff
@@ -0,0 +1,96 @@
+"""Support for Harmony Hub select activities."""
+import logging
+
+from homeassistant.components.select import SelectEntity
+from homeassistant.const import CONF_NAME
+from homeassistant.core import callback
+
+from .connection_state import ConnectionStateMixin
+from .const import ACTIVITY_POWER_OFF, DOMAIN, HARMONY_DATA
+from .data import HarmonyData
+from .subscriber import HarmonyCallback
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up harmony activities select."""
+    data = hass.data[DOMAIN][entry.entry_id][HARMONY_DATA]
+    _LOGGER.debug("creating select for %s hub activities", entry.data[CONF_NAME])
+    async_add_entities(
+        [HarmonyActivitySelect(f"{entry.data[CONF_NAME]} Activities", data)]
+    )
+
+
+class HarmonyActivitySelect(ConnectionStateMixin, SelectEntity):
+    """Select representation of a Harmony activities."""
+
+    def __init__(self, name: str, data: HarmonyData) -> None:
+        """Initialize HarmonyActivitySelect class."""
+        super().__init__()
+        self._name = name
+        self._data = data
+
+    @property
+    def name(self):
+        """Return the Harmony select activities name."""
+        return self._name
+
+    @property
+    def unique_id(self):
+        """Return the unique id."""
+        return f"{self._data.unique_id}_activities"
+
+    @property
+    def device_info(self):
+        """Return device info."""
+        return self._data.device_info(DOMAIN)
+
+    @property
+    def should_poll(self):
+        """Return that we shouldn't be polled."""
+        return False
```

Comment:
You can shorten a lot of these properties by using

Reply:
Should I do it for all properties like:
```python
        self._attr_name = name
        self._attr_unique_id = f"{self._data.unique_id}_activities"
        self._attr_device_info = self._data.device_info(DOMAIN)
        self._attr_should_poll = False
        self._attr_available = self._data.available
```

Reply:
https://github.com/home-assistant/core/pull/53943/commits/77e667246ee56b02f5c087a573e4614f863aa60f

Reply:
tests doesn't like that I delete `self._name` :sweat_smile: 
```FAILED tests/components/harmony/test_remote.py::test_connection_state_changes[pyloop] - AssertionError: assert False
FAILED tests/components/harmony/test_remote.py::test_connection_state_changes[pyloop] - AttributeError: 'HarmonyRemote' object has no attribute '_name'
FAILED tests/components/harmony/test_remote.py::test_remote_toggles[pyloop] - AssertionError: assert False```